spec/higher_level_api/integration/connection_spec.rb

```diff
@@ -61,6 +71,7 @@
     end
     c = MarchHare.connect(executor_factory: factory, network_recovery_interval: 0)
     c.close
+    c.instance_variable_set(:@was_explicitly_closed, false)
```

Comment:
to make these tests work I now override the internal instance var. Not ideal for sure. I also don't quite understand the intention behind the `close` + `automatically_recover` in these tests. Maybe you can shed some light on why